galaxy_ng/app/management/commands/dump_auth_config.py

```diff
@@ -0,0 +1,87 @@
+import json
+import os
+from django.core.management.base import BaseCommand
+from django.conf import settings
+from dynaconf.utils import upperfy
+
+
+class Command(BaseCommand):
+    KEYCLOAK_AUTH_KEYS = [
+        "SOCIAL_AUTH_KEYCLOAK_ACCESS_TOKEN_URL",
+        "SOCIAL_AUTH_KEYCLOAK_AUTHORIZATION_URL",
+        "SOCIAL_AUTH_KEYCLOAK_KEY",
+        "SOCIAL_AUTH_KEYCLOAK_PUBLIC_KEY",
+        "SOCIAL_AUTH_KEYCLOAK_SECRET",
+        "KEYCLOAK_PROTOCOL",
+        "KEYCLOAK_HOST",
+        "KEYCLOAK_PORT",
+        "KEYCLOAK_REALM",
+    ]
+
+    LDAP_AUTH_KEYS = [
+        "AUTH_LDAP_SERVER_URI",
+        "AUTH_LDAP_BIND_DN",
+        "AUTH_LDAP_BIND_PASSWORD",
+        "AUTH_LDAP_USER_SEARCH_BASE_DN",
+        "AUTH_LDAP_USER_SEARCH_SCOPE",
+        "AUTH_LDAP_USER_SEARCH_FILTER",
+        "AUTH_LDAP_GROUP_SEARCH_BASE_DN",
+        "AUTH_LDAP_GROUP_SEARCH_SCOPE",
+        "AUTH_LDAP_GROUP_SEARCH_FILTER",
+        "AUTH_LDAP_USER_ATTR_MAP",
+    ]
+
+    help = "Dump auth config data from database to a JSON file"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "output_file",
+            nargs="?",
+            type=str,
+            default="/auth_config.json",
+            help="Output JSON file path",
+        )
+
+    def format_config_data(self, type, keys, prefix):
+        config = {
+            "type": type,
+            "configuration": {}
+        }
+        for key in keys:
+            k = key
+            if prefix in key:
+                k = key[len(prefix):]
+            v = settings.get(upperfy(key), default=None)
+            config["configuration"].update({k: v})
+        return config
+
+    def handle(self, *args, **options):
+        try:
+            data = []
+
+            # Add Keycloak auth config
+            data.append(
+                self.format_config_data(
+                    "keycloak",
+                    self.KEYCLOAK_AUTH_KEYS,
+                    "SOCIAL_AUTH_KEYCLOAK_"),
+            )
+
+            # Add LDAP auth config
+            data.append(self.format_config_data("ldap", self.LDAP_AUTH_KEYS, "AUTH_LDAP_"))
+
+            # Define the path for the output JSON file
+            output_file = options["output_file"]
+
+            # Ensure the directory exists
+            os.makedirs(os.path.dirname(output_file), exist_ok=True)
+
+            # Write data to the JSON file
+            with open(output_file, "w") as f:
+                json.dump(data, f, indent=4)
+
+            self.stdout.write(
+                self.style.SUCCESS(f"Auth config data dumped to {output_file}")
+            )
+        except Exception as e:
+            self.stdout.write(self.style.ERROR(f"An error occurred: {str(e)}"))
```

Comment:
I would not save it directly to a file, (unless there is a specific requirement for doing it) 

Following the unix priciples, I would just output the JSON string to the stdout and let the caller redirect the stream to a file

```console
$ django-admin dump_auth_config > /path/to/file.json
$ django-admin dump_auth_config | jq "query I am interested in"
```

For this to work well, the command should not write anything else to the stdout, just the JSON and in case of error just

```py
except Exception as e:
    self.stderr.write(...)
    sys.exit(1)
```

Reply:
nevermind, reading the issue I see that saving to a file is a requirement.